generator/src/main/java/com/faforever/neroxis/generator/texture/BasicTextureGenerator.java

```diff
@@ -73,6 +75,28 @@ protected void setupTexturePipeline() {
                          .blur(2);
         texturesLowMask.setComponents(accentGroundTexture, accentPlateauTexture, slopesTexture, accentSlopesTexture);
         texturesHighMask.setComponents(steepHillsTexture, waterBeachTexture, rockTexture, accentRockTexture);
+
+        setupTerrainType(mapSize);
+    }
+
+    protected void setupTerrainType(int mapSize) {
+        terrainType.setSize(mapSize);
+        terrainType.startVisualDebugger();
+
+        Integer[] terrainTypes = map.getBiome().getTerrainMaterials().getTerrainTypes();
+        if (terrainTypes[0] == null) {
+            terrainType.add(1); // Default terrain type
+        } else {
+            terrainType.add(terrainTypes[0])
+                       .setToValue(accentGroundTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[1])
+                       .setToValue(accentPlateauTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[2])
+                       .setToValue(slopesTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[3])
+                       .setToValue(accentSlopesTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[4])
+                       .setToValue(steepHillsTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[5])
+                       .setToValue(waterBeachTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[6])
+                       .setToValue(rockTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[7])
+                       .setToValue(accentRockTexture.setSize(mapSize).copyAsBooleanMask(.5f), terrainTypes[8]);
+        }
```

Comment:
Meh I don't think I am a big fan of the null check. I think the initialization or setting the values would be best.
I don't think it should be the generators job to validate the input here. Rather correct input should just be given.

Reply:
How do I default to 1 values then if none are given? Or we could just skip that step alltogether and just add terrain types to all biomes. Could be all default values for starters

Reply:
You would just initalize the array in terrainMaterials as an array of all 1s. Or could just make it an int[] so that it is more obvious when a biome doesn't have any terrain types set.

Allthough I guess I have to check if the serializer we currently use would overwrite the value. If it does we would have to make a setter that checks in terrainMaterials

Reply:
seems like the serializer only overwrites the default values if he finds actual values in the json

Reply:
Then great we can just use the initialization to set default values

Reply:
I already pushed the change. Have a look if it is implement in the way you imagined